java/maven.embedder/external/apache-maven-3.9.6-httpclient-4.5.14-license.txt

```diff
@@ -0,0 +1,565 @@
+Name: httpclient 4.5.14, as part of Maven distribution
```

Comment:
The license for httpclient is not Apache 2.0 - it is Apache 2.0 + MPL 2.0, please see:
https://issues.apache.org/jira/browse/HTTPCLIENT-2317
https://issues.apache.org/jira/browse/MNG-8042

Reply:
Not sure I agree with your reading on this one.  Or theirs?  I would check with ASF legal on this, or if they did, because I don't see any reason that needs to be a dual licensed binary.  And ASF binaries should be ASL only.

Reply:
Note it is not a dual license. There are parts that are Apache 2.0, and parts (specifically the 'mozilla/public-suffix-list.txt' file) that are under the MPL 2.0. It is not really different from NetBeans' various 3rd party stuff.

I think this:
https://www.apache.org/legal/release-policy.html#license-file
is pretty clear.

Reply:
Note from MPL.

```
3.3. Distribution of a Larger Work

You may create and distribute a Larger Work under terms of Your choice,
provided that You also comply with the requirements of this License for
the Covered Software. If the Larger Work is a combination of Covered
Software with a work governed by one or more Secondary Licenses, and the
Covered Software is not Incompatible With Secondary Licenses, this
License permits You to additionally distribute such Covered Software
under the terms of such Secondary License(s), so that the recipient of
the Larger Work may, at their option, further distribute the Covered
Software under the terms of either this License or such Secondary
License(s).
```

As far as I can see that file is downloaded (as it should be under Cat B), so shouldn't be included in the source license file.  In binary form, it _could_ all be under ASL?

Reply:
For httpclient (both httpclient and httpcore) the license in the package is pure ALv2.0. I would suggest to stick with the upstream license and not pretend to know better. Once httpclient gets updated, I see no problem to follow though.

Reply:
I have no problem updating this here.

BUT: I refuse to chase down the potential intention of a packager through multiple layers of packages. `httpclient-4.5.14.jar` and `httpcore-4.4.16.jar` in the `apache-maven-3.9.6-bin.zip` both contain a LICENSE, both without MPL reference. It seems, that the maven central binaries are missing that part. This is an issue in apache maven. Next time this is updated, we will see a regression if it is not fixed in apache http client.

Reply:
The license file in the source tree is wrong.  The license file in the Maven binary might not be wrong.  MPL gives the right to distribute that artefact under ASL.

Reply:
I've removed this change to avoid the controversy.

Note that I've filled bugs against httpclient and Maven. So, while there's a potential for a regression on update, I think I did a lot to avoid that in the future. (And updating licenses is, sadly, not automatic when upgrading a library, so there's also an opposite risk - that Maven is fixed, but NetBeans is not.)

No sure I understand the argument of "they have it wrong, so it is OK for us to have it wrong". (While I hear Neil's idea about distributing under Apache 2.0 - noone has confirmed that would be what ASF could or would want to do.)